src/harness/testcases/WINNF_FT_S_REG_testcase.py

```diff
@@ -0,0 +1,262 @@
+        'fccId': device_a['fccId'],
+        'cbsdSerialNumber': device_a['cbsdSerialNumber'],
+        'airInterface': device_a['airInterface'],
+        'installationParam': device_a['installationParam']
+    }
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam']
+    }
+    conditionals_c = {
+        'cbsdCategory': device_c['cbsdCategory'],
+        'fccId': device_c['fccId'],
+        'cbsdSerialNumber': device_c['cbsdSerialNumber'],
+        'airInterface': device_c['airInterface'],
+        'installationParam': device_c['installationParam']
+    }
+    conditionals_d = {
+        'cbsdCategory': device_d['cbsdCategory'],
+        'fccId': device_d['fccId'],
+        'cbsdSerialNumber': device_d['cbsdSerialNumber'],
+        'airInterface': device_d['airInterface'],
+        'installationParam': device_d['installationParam']
+    }
+    conditionals = {
+        'registrationData': [
+            conditionals_a, conditionals_b, conditionals_c, conditionals_d
+        ]
+    }
+
+    # Inject FCC IDs
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    self._sas_admin.InjectFccId({'fccId': device_b['fccId']})
+    self._sas_admin.InjectFccId({'fccId': device_c['fccId']})
+    self._sas_admin.InjectFccId({'fccId': device_d['fccId']})
+
+    # Inject User IDs
+    self._sas_admin.InjectUserId({'userId': device_a['userId']})
+    self._sas_admin.InjectUserId({'userId': device_b['userId']})
+    self._sas_admin.InjectUserId({'userId': device_c['userId']})
+    self._sas_admin.InjectUserId({'userId': device_d['userId']})
+
+    self._sas_admin.PreloadRegistrationData(conditionals)
+
+    # Device 2 missing cbsdSerialNumber
+    del device_b['cbsdSerialNumber']
+
+    # Device 3 missing fccId
+    del device_c['fccId']
+
+    # Device 4 missing userId
+    del device_d['userId']
+
+    # Register devices
+    devices = [device_a, device_b, device_c, device_d]
+    request = {'registrationRequest': devices}
+    response = self._sas.Registration(request)
+    # Check registration response
+    self.assertTrue('cbsdId' in response['registrationResponse'][0])
+    self.assertEqual(
+        response['registrationResponse'][0]['response']['responseCode'], 0)
+    for x in range(1, 4):
+      self.assertEqual(
+          response['registrationResponse'][x]['response']['responseCode'], 102)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_REG_9(self):
+    """Blacklisted CBSD in Array Registration request (responseCode 101).
+
+    The response should be BLACKLISTED 101.
+    """
+
+    # Load Devices
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
```

Comment:
device B is a Cat B cbsd and it should not be able to Single-step register unles the CPI data is encoded and signed by a valid CPI.  For the purposes of this test we should only use Cat A CBSDs (easier to single-step- register in current code

Reply:
Done.